src/storage/src/source/postgres.rs

```diff
@@ -349,55 +350,77 @@ async fn postgres_replication_loop_inner(
     }
 }
 
-/// A helper struct build and produce transactions
-struct Transaction {
-    rows: Vec<(Row, Diff)>,
+/// A type that makes it easy to correctly send inserts and deletes.
+///
+/// Note: `RowSender::delete/insert` should be called with the same
+/// lsn until `close_lsn` is called, which must be called and awaited
```

Comment:
Should we add a `Drop` that either panics if there is pending work or tries to schedule a task on the current runtime?

Reply:
we could add a drop that panics if we are dropping, have pending work, _and are not already pancking_, but it seems pretty niche, especially to check `std::thread::panicking`. Note that dropping while still having pending work is not THAT unsafe, unless we somehow re-create this operator and assume an lsn is closed when its not, as the `RowSender` is `PostgresTaskInfo` level

Reply:
My suggestion was based off of your doc comment which says that `close_lsn` *must* be called and awaited before dropping. If that's not a strong requirement I'm fine not adding the `Drop` impl and maybe changing the doc comment to "should"